django/core/management/commands/shell.py

```diff
@@ -185,7 +186,10 @@ def get_namespace(self, **options):
             else:
                 module = None
                 name = path
+            if name in imported_names[module]:
+                continue
 
+            imported_names[module].add(name)
```

Comment:
Why do we need the additional `defaultdict`? 🤔  Something like this also fixes the issue:

```suggestion
            if (name, obj) not in auto_imports[module]:
                imported_names[module].add(name)
```

Reply:
@Hesham942 The point from @shangxiao (which I agree) is that there is no need to add a new variable `imported_names`.

The code should remove `imported_name` and roughly be:
```suggestion
            if (name, obj) not in auto_imports[module]:
                auto_imports[module].append((name, obj))
```

Reply:
Ok i will do that but everything else is ok?